projects/inc/saconfig.h

```diff
@@ -1,10 +1,19 @@
 #pragma once
 #include "hal.h"
 #include "encoder.h"
+#include "debug.h"
+#include <algorithm>
 
 /* sensor and actuator configuration constants */
 namespace sa {
 
+template <typename T>
```

Comment:
The clamp function would probably have a better place elsewhere, but I don't think we have a file for that now unless I forgot about it. saconfig is for sensor and actuator constants.